cmd/query/app/token_propagation_test.go

```diff
@@ -82,7 +83,11 @@ func runQueryService(t *testing.T, esURL string) *Server {
 	}))
 	f.InitFromViper(v, flagsSvc.Logger)
 	// set AllowTokenFromContext manually because we don't register the respective CLI flag from query svc
-	f.Options.Config.Authentication.BearerTokenAuthentication.AllowFromContext = true
+	if f.Options.Config.Authentication.BearerTokenAuthentication.HasValue() {
+		bearerAuth := f.Options.Config.Authentication.BearerTokenAuthentication.Get()
```

Comment:
Don't you want to deref the pointer here and take a copy? Otherwise you're changing the value returned from Get() as a pointer (not sure if that's a good idea to do). And if you were changing value via pointer anyway then probably re-assigning it again wouldn't be necessary.
```suggestion
		bearerAuth := *(f.Options.Config.Authentication.BearerTokenAuthentication.Get())
```